configure.ac

```diff
@@ -84,16 +86,25 @@ AC_MSG_RESULT([$H5_APIVER])
 
 #if version is unreleased then bump to next official release.
 
-AC_SUBST([H5_LIBVER_DIR])
 H5_MAJORVER=$(echo $H5_LIBVER | cut -f1 -d'.'  | sed -r 's/\./_/g')
 H5_MINORVER=$(echo $H5_LIBVER | cut -f2 -d'.'  | sed -r 's/\./_/g')
+H5_RELEASEVER=$(echo $H5_LIBVER | cut -f3 -d'.'  | sed -r 's/\./_/g')
+
+#used in fortran version checks
+H5_VERS_MAJOR=$H5_MAJORVER
+H5_VERS_MINOR=$H5_MINORVER
+H5_VERS_RELEASE=$H5_RELEASEVER
+
+AC_SUBST([H5_VERS_MAJOR])
+AC_SUBST([H5_VERS_MINOR])
+AC_SUBST([H5_VERS_RELEASE])
 
```

Comment:
CMake from the hdf5 library CMakeLists.txt file:

#-----------------------------------------------------------------------------
# parse the full version number from H5public.h and include in H5_VERS_INFO
#-----------------------------------------------------------------------------
file (READ ${HDF5_SRC_DIR}/H5public.h _h5public_h_contents)
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MAJOR[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_MAJOR ${_h5public_h_contents})
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_MINOR[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_MINOR ${_h5public_h_contents})
string (REGEX REPLACE ".*#define[ \t]+H5_VERS_RELEASE[ \t]+([0-9]*).*$"
    "\\1" H5_VERS_RELEASE ${_h5public_h_contents})

Reply:
I updated to using versioning in Fortran, thanks.